src/tcgwars/logic/impl/gen7/CosmicEclipse.groovy

```diff
@@ -4862,7 +4860,11 @@ public enum CosmicEclipse implements LogicCardInfo {
           playRequirement{
             assert my.deck : "Your deck is empty."
             assert bg.turnCount > 2 : "Cannot use this card during your first turn."
-            assert my.all.any{it.turnCount < bg.turnCount && it.lastEvolved < bg.turnCount} : "Cannot use this on Pokémon put into play this turn"
+            assertMyAll(info: "that weren't put into play this turn.", {
+              (it.turnCount < bg.turnCount && it.lastEvolved < bg.turnCount) &&
+              //TODO: Change this. Still can't learn if a Pokémon has a GX evolution, but this will at least filter obvious non-allowed cases for standard and expanded.
+              !(it.pokemonEX || it.pokemonBreak || it.pokemonGX || it.pokemonV || it.topPokemonCard.cardTypes.is(PRISM_STAR))
+              })
```

Comment:
Not a fan of this change, looks too complex.

Reply:
Left a TODO, but yeah it's not that big a deal. Will be fixed once that's implemented, for now it can remain a general check as before.